pkg/adaptation/result.go

```diff
@@ -386,6 +390,36 @@ func (r *result) adjustDevices(devices []*LinuxDevice, plugin string) error {
 	return nil
 }
 
+func (r *result) adjustCDIDevices(devices []*CDIDevice, plugin string) error {
+	if len(devices) == 0 {
+		return nil
+	}
+
+	// Notes:
+	//   CDI devices are opaque references, typically to vendor specific
+	//   devices. They get resolved to actual devices and potential related
+	//   mounts, environment variables, etc. in the runtime. Unlike with
+	//   devices, we only allow CDI devices references to be added to a
+	//   container, not removed. We pass them unresolved to the runtime and
+	//   have them resolved there. Also unlike with devices, we don't include
+	//   CDI device references in a creation requests. However, since there
```

Comment:
What is the scope at which these multiple references are checked.

It is, for example, perfectly fine for multiple containers to request the same device. In the case of a single container requesting the same device multiple times, this could also be seen as an ok use-case that could be solved by some kind of deduplication.

Reply:
With NRI and a single NRI request, (potential) multiplicity is present at the NRI plugin level, not the container level. With a few exceptions, a request is always about a single pod or a single container. However, there might be multiple plugins acting on the same request and applying modifications to the container via the NRI-provided interfaces. What we currently check is that only a single NRI plugin injects the same CDI device (using the same by fully qualified CDI device name) to the container. Multiple plugins attempting the same is flagged as a conflict/error.

Reply:
OK. Like I said, I'm happy to keep this as is for now, but at some point we may want to consider relaxing this restriction. In the end the CDI modifications are applied by the runtime and since CDI device names are locally unique, the end result should be the same.

Reply:
yeah some sort of allow plugin overrides (conflicts)